### PR TITLE
79 - [Blog Overview] Implement "Blog List" block with different filters

### LINKED ISSUE #79: [Blog Overview] Implement "Blog List" block with different filters
See
- https://www.moleculardevices.com/lab-notes/blog#viewall
- https://www.moleculardevices.com/lab-notes/blog#Cellular-Imaging-Systems
-etc

The "Blog List" block by default displays all block posts. Filters are provided via URL suffix.
The blog list data is provided via the index sheet, in a separate tab only filtering blog posts and sorted by publish date.

### PR DESCRIPTION
**Pull request by andreituicu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #79

Details:
 * Implemented Blog List which aggregates blog items and displays them as cards
 * On Desktop: Shows as a list
 * On Mobile: Shows as a carousel with infinite scrolling
 * Filtering works by changing the hash string in the url
 * Clicking the links on the page changes filtering without reload (they simply change the url hash string, no other event listeners is present on them, the list knows to filter itself on url hash change event)
 * Refactoring:
   * Move the blog card rendering to the blog.js/.css template so it can be reused in multiple components
   * Move the `summariseDescription` function in `scripts.js` to minimise critical path of file loadings

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.live/lab-notes/blog
- After: https://79-blog-list--moleculardevices--hlxsites.hlx.live/lab-notes/blog

### CODE REVIEW

**Review by jose-correia — COMMENTED** [review]
Looks good to me!

**Review by jose-correia — APPROVED** [review]

**Comment by mhaack** [comment]
I propose to rename and make the blocks more generic so they can be re-used for other card lists as well. I would not overload the `cards` block hence something like `dynamic cards`, `resource cards` or `cards from index` (I'm not creative with the naming). The same of the filter, maybe `cards filter`?

On the card list bock we get it a type like `dynamic cards(blog)`, `dynamic cards(products)`, `dynamic cards(applications)` of we use options as block rows. I'm thinking of providing the index URL as option 1 and a `type` option.

**Comment by andreituicu** [comment]
@mhaack I think we need 1 or 2 more implementation examples to be able to generalise it.
For example the citations list, which is not filterable, would not need almost any of the extra JS. The carousel.js would be enough and rendering it like a list is as simple as switching `flex-wrap: wrap` instead of `nowrap`.
And the filters I'm not sure yet how to make generic, because I haven't seen other examples. 
The current implementation is very particular for the blog items.

**Review by mhaack — APPROVED** [review]
As discussed we go with this, once working on the other card lists we can see what can be unified

File: scripts/scripts.js [hunk 1]
```diff
@@ -235,8 +253,8 @@ async function loadLazy(doc) {
   const element = hash ? doc.getElementById(hash.substring(1)) : false;
   if (hash && element) element.scrollIntoView();
 
-  const megaMenuModule = await import('../blocks/header/header-megamenu.js');
-  megaMenuModule.default(headerBlock);
+  // const megaMenuModule = await import('../blocks/header/header-megamenu.js');
+  // megaMenuModule.default(headerBlock);
```
[the remark this answers is not in the document]
Comment (andreituicu) [reply scripts/scripts.js]: Reverted in [fd7123c](https://github.com/hlxsites/moleculardevices/pull/232/commits/fd7123c54a42805bb2cb44098e750525d7e2e830)

**Pull request merged**